config.ru

```diff
@@ -4,13 +4,15 @@ require 'rack/rewrite'
 use Rack::ShowExceptions
 use Rack::ShowStatus
 
-# todo: look into using Refractor for these redirects
-
 # https://github.com/jtrupiano/rack-rewrite
 use Rack::Rewrite do
-  moved_permanently '/lessons/', '/lessons'
-  moved_permanently '/projects/', '/projects'
-  moved_permanently '/topics/', '/topics'
+
+  moved_permanently %r{^/(.*)/+$}, '/$1' # remove trailing /
+
+  # replace underscores with dashes
+  moved_permanently %r{^/(.*)_+(.*)$}, (lambda do |match, env|
```

Comment:
Is this RegExp going to work if the URL has mutiple path parts with underscrores?

For example: `http://bootcamp.burlingtoncodeacademy.com/lessons/client_side_coding/calling_apis`

What about replacing the text with a `string.scan(/^/(.*)_+(.*)$/).each { |m| m[0].gsub(/_+/, '-') }`

Reply:
The reason I used the proc is to do a gsub. The regex could be simplified but it matches any path with an underscore; the proc returns the full replacement.